File: 3private_encapsulation.cpp
#include<bits/stdc++.h>
using namespace std;

class BankAccount{
    private:
    // Encapsulate data (private Attributes)
        string name;
        double balance;

    public:
        BankAccount(string name, double balance =0.00){
            this->name= name;
            this->balance= balance;
        }

        void deposit(int amount){
            if(amount>0){
                this->balance+=amount;
                cout<<"Deposite Successfull"<<endl;
            }
            else
                cout<<"Invalid Deposit amount"<<endl;
        }

        double getBalance(){
            return this->balance;
        }
};

int main(){
    
    BankAccount user1("Rakib Khan");
    cout<<user1.getBalance()<<endl;
    user1.deposit(100);
    cout<<user1.getBalance();
    return 0;
}

File: 3private_encapsulation.py
class BankAccount:
    def __init__(self,nam, balance = 0.00)-> None:
        # Encapsulate data (private Attributes)
        self.__nam= nam
        self.__balance = balance
    
    def deposit(self, amount)->None:
        if amount > 0:
            self.__balance+=amount
            print("Deposite Successfull")
        else:
            print("Invalid Deposit amount")
    def getBalance(self)-> float:
        return self.__balance;

user1 = BankAccount("Rakib Khan")
print(user1.getBalance())
user1.deposit(500)
print(user1.getBalance())